### PR TITLE
Add a Dockerfile for running the tests

### PR DESCRIPTION
**Pull request by PirminTapken**
This way our venv is managed but separated from os.

### CODE REVIEW

**Comment by PirminTapken** [comment]
@bonko PTAL

**Comment by mlafeldt** [comment]
I recommend to mount the complete source code at runtime and keep the image as simple as possible.

I also suggest to create the image with `--force-rm` and the container with `--rm`.

**Comment by PirminTapken** [comment]
Now the code is mounted but the requirements.txt still copied in order to cache the pip install. The alternative would be to put the venv in code dir (which might be an option) but might not be usable on the host (wrong architecture / wrong python version). What do you think @mlafeldt ?

**Comment by mlafeldt** [comment]
That looks much better. This way, the image will only be updated if e.g. `requirements.txt` changes.

File: test.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+sudo docker build --force-rm -t pytrac/debian:stable .
+sudo docker run -rm -t -i  -v $PWD:/home/pytrac/data pytrac/debian:stable ./venv/bin/python ./setup.py test
```
Comment (mlafeldt) [inline test.sh]: Please don't use sudo, as this is not required on OSX.

This is how we do in the buildbox:
https://github.com/Jimdo/buildbox#quick-start

Also, `set -e` ftw. :)

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,8 @@
+FROM debian:stable
+MAINTAINER Pirmin Tapken <pirmin.tapken@jimdo.com>
+RUN apt-get update && apt-get install -y python2.6 python-virtualenv pandoc
+RUN mkdir -p /home/pytrac/data
+RUN virtualenv -p python2 /home/pytrac/venv
+WORKDIR /home/pytrac/data
+COPY ./requirements.txt /tmp
+RUN ./venv/bin/pip install -r /tmp/requirements.txt
```
Comment (Luzifer) [inline Dockerfile]: If I don't get it wrong you want to `../venv/bin/pip`…

File: test.sh (hunk 1 above)
Comment (Luzifer) [inline test.sh]: As the box is build exclusively you could use `ENTRYPOINT` for this. In that case you wouldn't have to care about the layout inside the container in your test script:

```docker
[...]
ENTRYPOINT ["/home/pytrac/venv/bin/python", "./setup.py", "test"]
```

Comment (PirminTapken) [inline test.sh]: @mlafeldt giving full docker access to normal linux users is a security issue as it essentially gives them root.

Comment (PirminTapken) [inline test.sh]: @Luzifer seems valid. Will do that next friday ;)

Comment (PirminTapken) [inline test.sh]: @mlafeldt a solution might be that linux users start the test.sh via sudo, but I don't like that either.

File: Dockerfile (hunk 2 above)
Comment (PirminTapken) [inline Dockerfile]: I would agree, but I don't understand why this works anyway :laughing:

**Comment by PirminTapken** [comment]
@Bonko @mlafeldt @Luzifer PTAL

**Comment by Luzifer** [comment]
Should work. LGTM

**Pull request merged**